src/loader.rs

```diff
@@ -0,0 +1,87 @@
+extern crate libloading;
+use self::libloading::{Library, Symbol};
+use super::driver::{Driver, DriverSchema};
+use std::collections::HashMap;
+
+pub type Result<T> = ::std::io::Result<T>;
+
+/// Trait implemented by anthing that wishes to load a driver at runtime.
+pub trait DriverLoader {
+  fn get_driver<'a>(&self) -> Option<&'a Driver>;
+  fn get_schema<'a>(&self) -> Option<&'a DriverSchema>;
```

Comment:
I'll think if there's a better way to model multiple features. Any suggestions? Hopefully we can find a good solution for this part of the code and the driver bits.